api/src/main/scala/types.scala

```diff
@@ -2274,7 +2274,7 @@ object BSONDocument {
    * // => { "foo": 1, "bar": 2 } : No "foo": 3
    * }}}
    */
-  def strict(elms: ElementProducer*): BSONDocument =
+  def strict(elms: ElementProducer*): BSONDocument with BSONStrictDocument =
```

Comment:
I don't think that's needed, as `BSONStrictDocument` just update the impl, not the API.

Reply:
Yes but it's nice for the library user to know that they're manipulating strict documents, and not just any documents. I want my functions to accept `BSONStrictDocument` arguments. So I can't call them with unsafe documents.

```
def saveStuff(doc: BSONDocument) = ???

save(BSONDocument("a" -> 3, "a" -> 4)) // oops!
```